### PR TITLE
*: introduce the cluster informer for decoupling the dependencies

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #5839.

### What is changed and how does it work?

Part of #6418.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/keyspace/keyspace.go [hunk 1]
```diff
@@ -26,8 +26,8 @@ import (
 	"github.com/pingcap/log"
 	"github.com/tikv/pd/pkg/id"
 	"github.com/tikv/pd/pkg/mcs/utils"
-	"github.com/tikv/pd/pkg/schedule"
 	"github.com/tikv/pd/pkg/schedule/labeler"
+	"github.com/tikv/pd/pkg/schedule/scheduling"
```
[the remark this answers is not in the document]
Comment (rleungx) [reply pkg/keyspace/keyspace.go:30]: Yes. but I don't have a better idea about the naming.

File: pkg/schedule/scheduling/cluster_informer.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2017 TiKV Project Authors.
+// Copyright 2020 TiKV Project Authors.
```
Comment (lhy1024) [inline pkg/schedule/scheduling/cluster_informer.go:1]: why change it?

**Review by lhy1024 — APPROVED** [review]

File: pkg/keyspace/keyspace.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (nolouch) [reply pkg/keyspace/keyspace.go:30]: How about change 
```
scheduling -> core or runtime 
schedulers -> strategies
```

**Review by JmPotato — APPROVED** [review]

File: pkg/keyspace/keyspace.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (rleungx) [reply pkg/keyspace/keyspace.go]: I change scheduling to core and keep the scheduler as it is.

**Review by nolouch — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Pull request merged**